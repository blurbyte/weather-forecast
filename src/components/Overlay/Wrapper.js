import styled from 'styled-components';

import { simpleFadeIn } from '../../styles/animations';

const Wrapper = styled.div`
  position: fixed;
  z-index: 9999;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0, 0, 0, 0.75);
  animation: ${simpleFadeIn} 0.15s ease-out forwards;
  opacity: 0;
`;

export default Wrapper;
