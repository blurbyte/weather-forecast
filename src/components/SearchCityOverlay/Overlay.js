import styled from 'styled-components';

import media from '../../styles/media';
import BaseOverlay from '../Overlay';

const Overlay = styled(BaseOverlay)`
  display: none;

  ${media.tablet`
    display: block;
  `};
`;

export default Overlay;
