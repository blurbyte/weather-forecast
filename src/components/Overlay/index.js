import React, { Component } from 'react';
import { createPortal } from 'react-dom';
import PropTypes from 'prop-types';

import Wrapper from './Wrapper';

class Overlay extends Component {
  constructor(props) {
    super(props);
    this.overlayContainer = document.createElement('div');
    document.body.appendChild(this.overlayContainer);
  }

  componentWillUnmount() {
    document.body.removeChild(this.overlayContainer);
  }

  render() {
    return createPortal(<Wrapper>{this.props.children}</Wrapper>, this.overlayContainer);
  }
}

Overlay.propTypes = {
  children: PropTypes.node
};

export default Overlay;
