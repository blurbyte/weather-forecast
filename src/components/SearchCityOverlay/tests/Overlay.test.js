import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Overlay from '../Overlay';
jest.mock('react-dom', () => ({
  createPortal: jest.fn(element => element)
}));

test('renders correctly', () => {
  const component = renderer.create(<Overlay>Portal Test</Overlay>);
  expect(component.toJSON()).toMatchSnapshot();
});
