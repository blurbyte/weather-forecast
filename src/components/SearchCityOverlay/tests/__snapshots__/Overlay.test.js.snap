// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c1 {
  position: fixed;
  z-index: 9999;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0,0,0,0.75);
  -webkit-animation: eMLfYp 0.15s ease-out forwards;
  animation: eMLfYp 0.15s ease-out forwards;
  opacity: 0;
}

.c0 {
  display: none;
}

@media (max-width:40em) {
  .c0 {
    display: block;
  }
}

<div
  className="c0 c1"
>
  Portal Test
</div>
`;
